google/cloud/forseti/common/util/email/email_factory.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2017 The Forseti Security Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Email factory to select connector"""
+
+from google.cloud.forseti.common.util import logger
+from google.cloud.forseti.common.util.email import sendgrid_connector
+from google.cloud.forseti.common.util.errors import InvalidInputError
+
+
+LOGGER = logger.get_logger(__name__)
+
+EMAIL_CONNECTOR_FACTORY = {
+    'sendgrid': sendgrid_connector.SendgridConnector
+}
+
+
+class EmailFactory(object):
+    """Email Factory to select connector."""
+
+    def __init__(self, notifier_config):
+        """Constructor for the email factory.
+
+        Args:
+            notifier_config (dict): Notifier configurations.
+        """
+        self.notifier_config = notifier_config
+        if notifier_config.get('email_connector'):
+            self.email_connector_config = (
+                notifier_config.get('email_connector'))
+
+    def get_connector(self):
+        """Gets the connector and executes it.
+
+        Returns:
+            object: Connector class
+
+        Raises:
+            InvalidInputError: Raised if invalid input is encountered.
+        """
+        # pylint: disable=logging-too-many-args
+        if not self.notifier_config:
+            LOGGER.exception('Notifier config is missing')
+            raise InvalidInputError(self.notifier_config)
+
+        if self.notifier_config.get('email_connector'):
```

Comment:
Again, asking if we can just check on `self.email_connector_config` directly here.  You will want to remove the `if` in the initi method.

```
self.email_connector_config = (
    notifier_config.get('email_connector'))
```

Reply:
Done